### DISCUSSION TITLE
How to manually resolve a service registered in IServiceCollection

### DISCUSSION THREAD

**Discussion by JohnHDev** [dotnet/maui#6887] [Q&A] [answered]
Hi all,
Trying to get to grips with the Maui way of doing things. If I register a service like so:
    builder.Services.AddSingleton<IMyRepository, MyRepository>();

How do I manually resolve that within another class without using constructor injection? I tried the following:
    var repo = serviceProvider.GetService<IMyRepository>();

where serviceProvider is of type IServiceProvider (which IS injected into the constructor of another class). But GetService just returns null.

**Comment by pictos** [comment, marked as answer]
@JohnHDev you can use this approach

https://github.com/davidortinau/WeatherTwentyOne/blob/main/src/WeatherTwentyOne/Services/ServiceExtensions.cs

**Comment by JohnHDev** [comment]
Thanks @pictos that looks good, although after rebooting and restarting VS, serviceProvider is now working as expected! :-0
